packages/synapse-react-client/src/components/SynapseTable/SynapseTableCell/SynapseTableCell.test.tsx

```diff
@@ -193,44 +202,120 @@ describe('SynapseTableCell tests', () => {
     rowVersionNumber: mockRowVersion,
   } satisfies Partial<SynapseTableCellProps>
 
-  it('renders an entity link with an ID', async () => {
-    renderTableCell({
-      ...tableCellProps,
-      rowData: [mockEntityLinkValue],
-      columnType: ColumnTypeEnum.ENTITYID,
-      columnValue: mockEntityLinkValue,
+  describe('PORTALS-2095, SWC-7235 - Special cases for ENTITYID columns', () => {
+    it('PORTALS-2095: renders an entity link for name column in EntityView', async () => {
+      server.use(
+        rest.get(
+          `${getEndpoint(
+            BackendDestinationEnum.REPO_ENDPOINT,
+          )}/repo/v1/entity/syn16787123`,
+          (req, res, ctx) => {
+            return res(ctx.status(200), ctx.json(mockFileViewEntity))
+          },
+        ),
+      )
+      renderTableCell({
+        ...tableCellProps,
+        rowData: ['My amazing project folder'],
+        columnName: 'name',
+        columnType: ColumnTypeEnum.STRING,
+        columnValue: 'My amazing project folder',
+      })
+
+      await screen.findByTestId('EntityLink')
+      expect(mockEntityLink).toHaveBeenCalledWith(
+        expect.objectContaining({
+          entity: String(mockRowId),
+          displayTextField: 'name',
+          showIcon: false,
+        }),
+        expect.anything(),
+      )
```

Comment:
Existing test behavior is unchanged, new assertion added to validate `EntityLink` passed props